### PR TITLE
Re-enable ping timeout behaviour for 3x branch

### PR DESCRIPTION
**Pull request by TorMFinn**
After this commit https://github.com/socketio/socket.io-client-cpp/commit/ec4d540ad54593604ac2091e67ffc2a6d9a00db6 the ping timeout behavior was broken.

Fixes:
* Re-enable ping timeout handling.
* Remove ping function that was unused.

### CODE REVIEW

**Comment by jmigual** [comment]
Hi @TorMFinn thanks for your contribution! If you could answer my question and change it (if necessary) then this PR will be ready to go 😊

File: src/internal/sio_client_impl.cpp [hunk 1]
```diff
@@ -484,11 +461,6 @@ namespace sio
     
     void client_impl::on_message(connection_hdl, client_type::message_ptr msg)
     {
-        if (m_ping_timeout_timer) {
-            asio::error_code ec;
-            m_ping_timeout_timer->expires_from_now(milliseconds(m_ping_timeout),ec);
-            m_ping_timeout_timer->async_wait(std::bind(&client_impl::timeout_pong, this, std::placeholders::_1));
-        }
         // Parse the incoming message according to socket.IO rules
```
[the remark this answers is not in the document]
Comment (TorMFinn) [reply src/internal/sio_client_impl.cpp:464]: I could re-add this, but I'm not sure it's "correct" behaviour to essentially treat every received message from the server as a "ping". I have checked the client implementation for Java script, and it seems like that client only updates the timer on a ping package also.

https://github.com/socketio/socket.io-client/blob/master/dist/socket.io.js

**Comment by TorMFinn** [comment]
Hi @jmigual I have commented on your question :)

File: src/internal/sio_client_impl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmigual) [reply src/internal/sio_client_impl.cpp:464]: You are right. The server is the one periodically sending ping packets that the client must reply. For more info see: https://socket.io/docs/v4/how-it-works/#disconnection-detection

**Pull request merged**